### DISCUSSION TITLE
Marker interface

### DISCUSSION THREAD

**Discussion by gitToad** [eidolonFIRE/xcNav_web-DEPRECATED-#9] [Ideas]
This needs more discussion from a functional perspective:

how do we create markers in the UI 
- marker color, icon, popup text etc.
- location obviously

How to edit markers once they are created (do we need the concept of selection and  selection feedback?) Enable selection of just one or multiple markers ?

delete markers ?

Allow creation of polylines and polygons as well as markers (for parity with Google Maps/Earth generated KMZ overlays ?) Or feature creep ?

What should we be able to do with markers ?
- distance (and time and heading) to marker from my current location could be automatic in popups
- do we need measurements between markers as well or just between myself and the marker ?

Caleb suggested being able to  optionally share a marker with the other pilots, it will pop up on their map. This would be an explicit action, unless you choose to share a marker, they will appear just in your map.

Somewhat overlapping: flight Plan overlays typically have a polyline that represents the flight path with waypoints. This is currently not enforced by the creation tools (ie you can make multiple polylines in the Google tools and have waypoints on them or not etc.)   It might be nice to have automatic waypoints in the app, so that it can tell you your next heading, distance, time to arrival etc. And like FlySkyHy, you might want to manually switch to an earlier or later waypoint. We could either do this by just selecting markers visually (already covered by the features above) or add our own standards to KMZ files and a way to extract waypoints (vs just general POIs) and an interface to select next/previous waypoint. TBD  !

**Comment by eidolonFIRE** [comment]
Had some ideas on how to present waypoints.
In my experience of planning routes, I want to have "alternates" and "paths". Alternates shouldn't be calculated in remaining distance, and paths can be rigid or visual only.
![image](https://user-images.githubusercontent.com/11761237/120690994-bfdd8a80-c45a-11eb-9e70-13987c1b3940.png)

**Reply by gitToad** [reply]
Time/miles to arrival shown for each guided & visual waypoint (along flightpath, ie not as the crow flies) ?  (any marker on the map usually displays as the crow flies time/miles in the popup when clicked?)

in the KMZs, do we designate one polyline as "the blessed flightpath" (there can be more than one polyline in a kmz)
by name ? by color ?

in the KMZs, markers are not related to any other objects like polylines. For our purposes the guided, visual waypoints need to be "along the flightpath" - need some way to associate them. The actual distance and ordering along the path is another open question, could do ordering by number scheme as prefix in name - huge pain if edits are required.  (eg you had way points 1..15 and now you want to add a waypoint between 2 and 3 - you need to edit 3..15 number prefixes to be 4..16 ugh)

How about using folder mechanism in KMZ, ie **one folder is called flightpath**
inside that, exactly one polyline (or error on import), which will be interpreted as the flightpath and receive special treatment. Such as being able to be displayed in a dialog like Caleb shows in his picture.

All markers inside that folder are used for guide and visual waypoints, in order of their appearance. Flight plan author has to make sure they actually are ordered that way in the KMZ. 

Those markers are assumed to be pretty close to the flightpath (there is no concept of "on path" in KMZ) 
Inside the app, they would be projected to the nearest point on the path and displayed as such (ie always exactly on the path) That location would also be used for parametric distance along the path.
